src/DGtal/geometry/curves/parametric/UglyNaiveParametricCurveDigitizer3D.ih

```diff
@@ -0,0 +1,312 @@
+    if ( digitalCurve.size() == 0 )
+    {
+      digitalCurve.push_back ( *bbegin );
+      it++;
+      if ( metaData )
+        metaDataContainter.push_back ( weights[*bbegin] );
+    }
+
+    for (; it != bend; it++ )
+    {
+      // search for better candidates i.e. search for 26-connected points of higher scores in the k-neighborhood.
+      for ( KConstIter s = { it + 1, 0 };  s.jt != bend && s.k < K_NEXT; s.jt++ )
+      {
+        if ( is26Connected ( digitalCurve.back(), *s.jt ) && weights[*s.jt].second >= weights[*it].second )
+        {
+          it = s.jt;
+          s.k = 0;
+        }
+        else if ( ! is26Connected ( digitalCurve.back(), *s.jt ) )
+          s.k++;
+      }
+
+      digitalCurve.push_back ( *it );
+      if ( metaData )
+        metaDataContainter.push_back( weights[*it] );
+    }
+  }
+
+
+  template <typename T>
+  inline
+  void UglyNaiveParametricCurveDigitizer3D<T>::flashBuffers ( Buffer & buffer, DataInfo & weights )
+  {
+    syncData ( buffer.begin(), buffer.end() - K_NEXT, weights );
+    // keep the last few points for re-evaluation or because they may not be well evaluated yet.
+    for ( auto it = buffer.begin(); it != buffer.end() - K_NEXT; it++ )
+      weights.erase ( *it );
+    buffer.erase ( buffer.begin(), buffer.end() - K_NEXT );
+  }
+
+  template <typename T>
+  inline
+  void UglyNaiveParametricCurveDigitizer3D<T>::updateMetaDate ( const Point & p, const RealPoint & pc,
+                                                                DataInfo & weights, long double t )
+  {
+    if ( metaData )
+    {
+      if ( weights[p].second == 1 )
+        weights[p].first = t;
+      else if ( (p - pc).norm() < (curve->x ( weights[p].first ) - p).norm() )
+        weights[p].first = t;
+    }
+  }
+
+   template <typename T>
+  inline
+  void UglyNaiveParametricCurveDigitizer3D<T>::cleanClosedPart ( )
+  {
+    bool isClosed = false;
+    auto it = digitalCurve.end () - K_NEXT;
+    auto jt = digitalCurve.begin () + K_NEXT;
+    for ( ; it != digitalCurve.end (); it++ )
+     for (; jt != digitalCurve.begin(); jt--)
+      if ( is26Connected ( *jt, *it ) )
+      {
+        isClosed = true;
+        break;
+      }
+    if ( isClosed && std::distance ( it, digitalCurve.end () ) > 1 )
+    {
+      if ( metaData )
+        metaDataContainter.erase ( metaDataContainter.end ( ) + std::distance ( it, digitalCurve.end ( ) ) + 1, metaDataContainter.end ( ) );
+      digitalCurve.erase ( it + 1,  digitalCurve.end ( ) );
+    }
+    if ( isClosed && jt != digitalCurve.begin() )
+    {
+      if ( metaData )
+        metaDataContainter.erase ( metaDataContainter.begin (), metaDataContainter.begin () + std::distance ( digitalCurve.begin (), jt )  );
+      digitalCurve.erase ( digitalCurve.begin ( ), jt );
+    }
+  }
+
+  template <typename T>
+  inline
+  void UglyNaiveParametricCurveDigitizer3D<T>::cleanCurve ( )
+  {
+    struct KIter { typename DigitalCurve::iterator jt; unsigned int k; };
```

Comment:
I got a warning here with GNU 8.1 since the `KIter` struct is already defined in `UglyNaiveParametricCurveDigitizer3D`: https://github.com/DGtal-team/DGtal/pull/1339/files#diff-3e3c950b82427f6070ab12d771e8e59bR125